### PR TITLE
Implement Account creation

### LINKED ISSUE #4: Implement Account creation
Implement the grpc service and store users in the database

### PR DESCRIPTION
**Pull request by skamoen**
fixes #4

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1 @@
+go.sum
```
Comment (martijnjanssen) [inline .gitattributes]:
```suggestion
go.sum -diff
```

File: internal/pkg/database/user.go [hunk 2]
```diff
@@ -1,22 +1,24 @@
...
+import "context"
+
 type User struct {
 	Email    string `db:"email"`
 	Password string `db:"password"`
 }
 
-func UserAdd(db Queryer, email string, password string) error {
-	_, err := db.Exec("INSERT INTO users (email, password) VALUES ($1, $2)", email, password)
+func UserAdd(ctx context.Context, db Queryer, email string, password string) error {
```
Comment (skamoen) [inline internal/pkg/database/user.go]: Let's rename `password` to `passwordHash` to make it clearer it's not cleartext

File: internal/pkg/database/queryer.go [hunk 3]
```diff
@@ -1,12 +1,16 @@
...
 )
 
 // Shared interface over both sqlx.Db and sqlx.Tx to enable
 // querying on both without considering the underlying type
 type Queryer interface {
 	Exec(string, ...interface{}) (sql.Result, error)
+	ExecContext(context.Context, string, ...interface{}) (sql.Result, error)
+
 	Get(interface{}, string, ...interface{}) error
+	GetContext(context.Context, interface{}, string, ...interface{}) error
```
Comment (skamoen) [inline internal/pkg/database/queryer.go:15]: I get the difference between the different funcs, but what would we lose by dropping the context-less variant here? If you don't want to pass the context you can just pass `context.Background()`

File: internal/pkg/database/user.go [hunk 4]
```diff
@@ -1,22 +1,24 @@
 package database
 
+import "context"
```
Comment (skamoen) [inline internal/pkg/database/user.go:3]: The fact that we're requiring the context here indicates the context-less variants aren't used

**Review by skamoen — COMMENTED** [review]
Github won't let me approve "my" PR, but I approve :D

File: internal/pkg/database/queryer.go (hunk 3 above)
Comment (martijnjanssen) [reply internal/pkg/database/queryer.go:15]: You're right, that would also work just fine. Anyways, enforcing the use of context with a deadline is always beneficial. I'll drop the context-less version!

Comment (martijnjanssen) [reply internal/pkg/database/queryer.go:15]: Upon further inspection, these are actually used. They are used in the migrations, where there is no deadline involved. We could drop it, but I'm not sure what the benefit would be apart from just 2 methods less.

**Review by martijnjanssen — APPROVED** [review]
Approving following mutual agreement of the changes.

**Pull request merged**